src/prefect/experimental/workers/base.py

```diff
@@ -98,10 +104,48 @@ def json_template(cls) -> dict:
         return configuration
 
     @staticmethod
-    def _apply_variables(job_config, variables):
-        """Apply variables to configuration template."""
-        job_config.update(variables)
-        return job_config
+    def _apply_variables(job_config: Dict[str, str], variables: Dict[str, Any]):
+        """
+        Apply variables to a job configuration template by interpolating variables into the placeholders in the
+        template.
+
+        If a value in the job configuration contains only a single placeholder, the value will be fully replaced
+        with the value.
+
+        If a value in the job configuration contains text before and after a placeholder or there are multiple
+        placeholders, the placeholders will be replaced with the corresponding variable values.
+
+        Args:
+            job_config: The job configuration portion of a base job template to apply variables to
+            variables: The variables to apply to the job configuration
+
+        Returns:
+            The job configuration with variables applied
+        """
+        variable_capture_regex = re.compile(r"({{\s*(\w+)\s*}})")
+        applied_config = {}
+        for key, value in job_config.items():
```

Comment:
might be worthwhile to anticipate `None` for `job_config` and do something like
```suggestion
        for key, value in (job_config or {}).items():
```